reference/multigrid/pmis_kernels.cpp

```diff
@@ -32,7 +33,45 @@ void compute_strong_dep_row(std::shared_ptr<const DefaultExecutor> exec,
                             remove_complex<ValueType> strength_threshold,
                             IndexType* sparsity_rows)
 {
-    GKO_NOT_IMPLEMENTED;
+    using rc = remove_complex<ValueType>;
+
+    const auto nrow = csr->get_size()[0]; 
+    const auto row_ptrs = csr->get_const_row_ptrs();   
+    const auto col_idxs = csr->get_const_col_idxs();   
+    const auto vals     = csr->get_const_values();  
+
+    sparsity_rows[0] = IndexType{0};
+
+    for (auto i = 0; i < nrow; ++i) {
+        
+        sparsity_rows[i+1] = sparsity_rows[i];
```

Comment:
because we have `prefix_sum_nonnegative` after this, they will build up the offset for csr.
each row only needs to record the number for its own row.